compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -93,21 +121,26 @@ fn adt_sized_constraint<'tcx>(
     let def = tcx.adt_def(def_id);
 
     if !def.is_struct() {
-        bug!("`adt_sized_constraint` called on non-struct type: {def:?}");
+        bug!("`adt_sizedness_constraint` called on non-struct type: {def:?}");
     }
 
     let tail_def = def.non_enum_variant().tail_opt()?;
     let tail_ty = tcx.type_of(tail_def.did).instantiate_identity();
 
-    let constraint_ty = sized_constraint_for_ty(tcx, tail_ty)?;
+    let constraint_ty = sizedness_constraint_for_ty(tcx, sizedness, tail_ty)?;
 
-    // perf hack: if there is a `constraint_ty: Sized` bound, then we know
+    // perf hack: if there is a `constraint_ty: {Meta,Pointee,}Sized` bound, then we know
     // that the type is sized and do not need to check it on the impl.
-    let sized_trait_def_id = tcx.require_lang_item(LangItem::Sized, None);
+    let lang_item = match sizedness {
+        SizedTraitKind::Sized => LangItem::Sized,
+        SizedTraitKind::MetaSized => LangItem::MetaSized,
+        SizedTraitKind::PointeeSized => LangItem::PointeeSized,
+    };
```

Comment:
make this a method on `SizedTraitKind` and reuse it

Reply:
It's a little tricky to do this because `SizedTraitKind` is in `type_ir` so only has `TraitSolverLangItem`, and the mapping between `LangItem` and `TraitSolverLangItem` isn't public, so I just added a `SizedTraitKind::require_lang_item` method that returned the `DefId` and took an `Interner`. I only found this snippet of code to use the function at though, from a couple quick greps of `match sizedness` or `match sized_trait_kind`, I don't get the `LangItem` elsewhere.

Reply:
yea makes sense to just not do anything then, thx for checking